cli/stork-publisher-agent/registry.go

```diff
@@ -0,0 +1,53 @@
+package stork_publisher_agent
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+type RegistryClient struct {
+	baseUrl   string
+	authToken AuthToken
+}
+
+func NewRegistryClient(baseUrl string, authToken AuthToken) *RegistryClient {
+	return &RegistryClient{
+		baseUrl:   baseUrl,
+		authToken: authToken,
+	}
+}
+
+func (c *RegistryClient) GetBrokersForPublisher(publisherKey PublisherKey) (map[BrokerPublishUrl]map[AssetId]struct{}, error) {
+	brokerEndpoint := c.baseUrl + "/v1/registry/brokers"
+	queryParams := url.Values{}
+	queryParams.Add("publisher_key", string(publisherKey))
+	authHeader := http.Header{"Authorization": []string{"Basic " + string(c.authToken)}}
+	response, err := RestQuery("GET", brokerEndpoint, queryParams, nil, authHeader)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get broker list: %v", err)
+	}
+	var brokers []BrokerConnectionConfig
+	err = json.Unmarshal(response, &brokers)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal broker list: %v", err)
+	}
+
+	// combine all configs into a single asset map per url
```

Comment:
technically we could have multiple entries in the publisher connection table for a given Broker (since there could be multiple agreements with different payment terms for different assets). We merge all the results into a single map from broker URL to the set of assets for that URL